opm/autodiff/BlackoilModelParameters.cpp

```diff
@@ -99,6 +103,10 @@ namespace Opm
         use_multisegment_well_ = false;
         matrix_add_well_contributions_ = false;
         preconditioner_add_well_contributions_ = false;
+        do_adjoint_=false;
+        use_amgcl_ = false;
+        use_amgcl_drs_ = true;//only used if amgcl is true
+        use_umfpack_ = false;
```

Comment:
note: this file is only used by the legacy simulators. adding parameters that do not do anything is potentially confusing? maybe some classes need to be split into a legacy and a non-legacy version to avoid this...